### ISSUE TITLE
Plugin throws `java.lang.NoSuchMethodError` when used with Android Gradle Plugin v3.6.0-rc01

### ISSUE THREAD

**Issue by nisrulz** [Archinamon/android-gradle-aspectj#103]
Error:

```> Failed to notify project evaluation listener.
   > com.android.build.gradle.BasePlugin.getVariantManager()Lcom/android/build/gradle/internal/VariantManager;
   > com.android.build.gradle.BasePlugin.getVariantManager()Lcom/android/build/gradle/internal/VariantManager;
```

StackTrace:

```
Caused by: java.lang.NoSuchMethodError: com.android.build.gradle.BasePlugin.getVariantManager()Lcom/android/build/gradle/internal/VariantManager;
        at com.archinamon.utils.VariantUtilsKt.getVariantScopes(VariantUtils.kt:105)
        at com.archinamon.utils.VariantUtilsKt.getVariantDataList(VariantUtils.kt:101)
        at com.archinamon.plugin.PluginSetupKt.prepareVariant(PluginSetup.kt:52)
        at com.archinamon.plugin.PluginSetupKt.access$prepareVariant(PluginSetup.kt:1)
        at com.archinamon.plugin.PluginSetupKt$configProject$1.invoke(PluginSetup.kt:27)
        at com.archinamon.plugin.PluginSetupKt$configProject$1.invoke(PluginSetup.kt)
        at com.archinamon.plugin.PluginSetupKt$whenEvaluated$1.invoke(PluginSetup.kt:125)
        at com.archinamon.plugin.PluginSetupKt$whenEvaluated$1.invoke(PluginSetup.kt)
        at com.archinamon.plugin.PluginSetupKt$sam$i$org_gradle_api_Action$0.execute(PluginSetup.kt)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction$1.lambda$run$0(DefaultListenerBuildOperationDecorator.java:152)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.reapply(DefaultUserCodeApplicationContext.java:60)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction$1.run(DefaultListenerBuildOperationDecorator.java:152)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction.execute(DefaultListenerBuildOperationDecorator.java:149)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:92)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:80)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:42)
        at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:231)
        at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:150)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:58)

```

**Comment by nisrulz** [comment]
AspectJ Plugin v4.0.1
Get error with AGP v3.6.0-rc01

Works fine with AGP v3.5.3

**Comment by Archinamon** [comment]
4.0.1 plugin supports only gradle 6.0+ and AGP 3.5.+
It's binary incompatible with older versions.

**Comment by nisrulz** [comment]
I would like to help, but I have not worked on a plugin before. I did fix the problem on my side, so I will try opening a PR.

**Comment by molpoo** [comment]
@nisrulz How did you manage to fix this problem ?
Are you also on Android Studio 3.6 with AGP 3.6+ and gradle 5.6.4 ?

A PR would be appreciated, if you don't feel confident about create it, I can help.

**Comment by DDeg** [comment]
Pre-build jar from pull-request. Add file to project location and connect to Gradle via

_classpath files('project/path/local_built_library.jar')_

[android-gradle-aspectj-4.1.0.jar.zip](https://github.com/Archinamon/android-gradle-aspectj/files/4420040/android-gradle-aspectj-4.1.0.jar.zip)

**Comment by nisrulz** [comment]
> Pre-build jar from pull-request. Add file to project location and connect to Gradle via
> 
> _classpath files('project/path/local_built_library.jar')_
> 
> instead of maven-published plugin.
> 
> [android-gradle-aspectj-4.1.0.jar.zip](https://github.com/Archinamon/android-gradle-aspectj/files/4420040/android-gradle-aspectj-4.1.0.jar.zip)
> 
> Tested with 3.6.1 plugin / Gradle 6.0.1

@DDeg Thank you for this. 

Have you considered using Jitpack to publish it from your github repo directly. Would allow to avoid having to validate the jar for security reasons. 

The process is very simple: You just need to add a tag. Goto Jitpack, select the tag and it will publish the jar file.

**Comment by Archinamon** [comment]
Merged, deployed as 4.1.0!
Many thanks for PR!

**Comment by oakkitten** [comment]
i'm getting a similar error:

    > gradlew --stacktrace

    FAILURE: Build failed with an exception.

    * What went wrong:
    A problem occurred configuring project ':app'.
    > Failed to notify project evaluation listener.
       > com.android.build.gradle.internal.VariantManager.getVariantScopes()Ljava/util/List;
       > com.android.build.gradle.internal.VariantManager.getVariantScopes()Ljava/util/List;

    * Exception is:
    org.gradle.api.ProjectConfigurationException: A problem occurred configuring project ':app'.
            at org.gradle.configuration.project.LifecycleProjectEvaluator.wrapException(LifecycleProjectEvaluator.java:75)
            at org.gradle.configuration.project.LifecycleProjectEvaluator.addConfigurationFailure(LifecycleProjectEvaluator.java:68)
            ...
    Caused by: org.gradle.internal.event.ListenerNotificationException: Failed to notify project evaluation listener.
            at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:86)
            at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:346)
            ...
    Cause 1: java.lang.NoSuchMethodError: com.android.build.gradle.internal.VariantManager.getVariantScopes()Ljava/util/List;
            at com.archinamon.utils.VariantUtilsKt.getVariantScopes(VariantUtils.kt:105)
            at com.archinamon.utils.VariantUtilsKt.getVariantDataList(VariantUtils.kt:101)
            at com.archinamon.plugin.PluginSetupKt.prepareVariant(PluginSetup.kt:53)
            at com.archinamon.plugin.PluginSetupKt.access$prepareVariant(PluginSetup.kt:1)
            ...
    Cause 2: java.lang.NoSuchMethodError: com.android.build.gradle.internal.VariantManager.getVariantScopes()Ljava/util/List;
            at com.archinamon.utils.VariantUtilsKt.getVariantScopes(VariantUtils.kt:105)
            at com.archinamon.utils.VariantUtilsKt.getVariantDataList(VariantUtils.kt:101)
            at com.archinamon.api.transform.AspectJTransform$prepareProject$1.execute(AspectJTransform.kt:37)
            at com.archinamon.api.transform.AspectJTransform$prepareProject$1.execute(AspectJTransform.kt:23)
            ...

windows; gradle 6.5; android gradle plugin 4.1.0
